Modules/+Sources/Cobolt_PB.m

```diff
@@ -0,0 +1,181 @@
+        function blackout(obj)
+            obj.diode_on = false;
+        end
+        
+        function delete(obj)
+            delete(obj.listeners)
+        end
+        
+        function val = set_power(obj, val, ~)
+            if obj.isConnected && ~isnan(val)
+                errorIfNotOK(obj.serial.com('Cobolt', 'slmp', val));    % Set laser modulation power (mW)
+            else
+                val = NaN;
+            end
+        end
+        function val = set_diode_on(obj, val, ~)
+            if obj.isConnected()
+                if val
+                    errorIfNotOK(obj.serial.com('Cobolt', '@cobas', 0));    % No autostart
+                    errorIfNotOK(obj.serial.com('Cobolt', 'em'));           % Enter Modulation Mode
+                    errorIfNotOK(obj.serial.com('Cobolt', 'l1'));
+                else
+                    errorIfNotOK(obj.serial.com('Cobolt', 'l0'));           % Laser off
+                end
+            else
+                val = false;
+            end
+        end
+        
+        function val = get_temperature(obj, ~)
+            val = obj.com('rbpt?');
+        end
+        function val = get_diode_sn(obj, ~)
+            val = obj.com('sn?');
+        end
+        function val = get_firmware_ver(obj, ~)
+            val = obj.com('ver?');
+        end
+        function val = get_diode_age(obj, ~)
+            val = obj.com('hrs?');
+        end
+        
+        function val = com(obj, str, varargin)
+            if obj.isConnected()
+                val = obj.serial.com('Cobolt', str);
+            else
+                val = NaN;
+            end
+        end
+        function tf = isConnected(obj)
+            tf = ~strcmp('No Server', obj.cobolt_host) && strcmp('OK', obj.serial.com('Cobolt', '?'));
+            
+            if ~tf
+                obj.set_cobolt_host(obj,'No Server')
+            end
+        end
+        
+        function val = set_pb_host(obj,val,~) %this loads the pulseblaster driver
+            if strcmp('No Server',val)
+                obj.PulseBlaster = [];
+                delete(obj.listeners)
+                obj.source_on = false;
+                return
+            end
+            err = [];
+            try
+                obj.PulseBlaster = Drivers.PulseBlaster.StaticLines.instance(val); %#ok<*MCSUP>
+                obj.source_on = obj.PulseBlaster.lines(obj.PB_line);
+                delete(obj.listeners)
+                obj.listeners = addlistener(obj.PulseBlaster, 'running', 'PostSet', @obj.isRunning);
+                obj.isRunning;
+            catch err
+                obj.PulseBlaster = [];
+                delete(obj.listeners)
+                obj.source_on = false;
+                val = 'No Server';
+            end
+            if ~isempty(err)
+                rethrow(err)
+            end
+        end
+        function val = set_cobolt_host(obj,val,~) %this loads the hwserver driver
+            if strcmp('No Server', val)
```

Comment:
I think it is worth deleting `obj.serial` every time this is called (it isn't singleton)